### PR TITLE
Adding phasefieldx package

### PR DESCRIPTION
**Pull request by CastillonMiguel**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by CastillonMiguel** [comment]
Ready for review, @conda-forge/help-python

**Comment by CastillonMiguel** [comment]
@conda-forge-admin, please ping team

File: recipes/phasefieldx/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,57 @@
+{% set name = "phasefieldx" %}
+{% set version = "0.0.3" %}
```
Comment (ocefpaf) [inline recipes/phasefieldx/meta.yaml:2]: There is a 0.0.4 on PyPI. Is there a reason to not use the source there?

File: recipes/phasefieldx/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+source:
+  url: https://github.com/CastillonMiguel/phasefieldx/archive/v{{ version }}.tar.gz
+  sha256: f85113793ba1f199647f715a94ee5f36cc96f6288aef0dbe227812e8e7b70402
+
+build:
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
+  number: 0
+  skip:
+    - win-64
```
Comment (ocefpaf) [inline recipes/phasefieldx/meta.yaml:17]:
```suggestion
  skip: True  # [win]
```

File: recipes/phasefieldx/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    - fenics-dolfinx ==0.8.0
+    - gmsh
+    - matplotlib
+    - mpi4py
+    - mpich
+    - pandas
+    - petsc4py
+    - numpy
+    - pyvista
+    - time
```
Comment (ocefpaf) [inline recipes/phasefieldx/meta.yaml:35]: I cannot see any of those dependencies in the metadata: https://github.com/CastillonMiguel/phasefieldx/blob/v0.0.3/pyproject.toml#L13
 
```suggestion
```

If they are rally needed, can you open an issue upstream?

File: recipes/phasefieldx/meta.yaml (hunk 1 above)
Comment (CastillonMiguel) [reply recipes/phasefieldx/meta.yaml]: Thanks for pointing that out! There was no specific reason for not using version 0.0.3. I’ve just updated the project to version 0.0.4    https://github.com/conda-forge/staged-recipes/pull/27848/commits/a5096ea59bf373dc04ed717a5ea5182433c57c54

File: recipes/phasefieldx/meta.yaml (hunk 3 above)
Comment (CastillonMiguel) [reply recipes/phasefieldx/meta.yaml]: The dependencies are indeed necessary, and I’ve specified some of them in the installation instructions. I will make sure to include all of them

File: recipes/phasefieldx/meta.yaml (hunk 2 above)
Comment (CastillonMiguel) [reply recipes/phasefieldx/meta.yaml:17]: I don’t know why, but I got an error when I changed to the suggestion, so I reverted to the last version.

File: recipes/phasefieldx/meta.yaml (hunk 3 above)
Comment (CastillonMiguel) [reply recipes/phasefieldx/meta.yaml]: https://github.com/CastillonMiguel/phasefieldx/commit/3395d76abde845fbbaab7a98b108ca8e05abde74

File: recipes/phasefieldx/meta.yaml (hunk 2 above)
Comment (ocefpaf) [reply recipes/phasefieldx/meta.yaml:17]: What error? This seems to be skipping everything.

Comment (CastillonMiguel) [reply recipes/phasefieldx/meta.yaml]: I have added the suggested changes again, but I am getting this error (if I remove noarch, I also get errors)

For recipes/phasefieldx/meta.yaml:

noarch packages can't have skips with selectors. If the selectors are necessary, please remove noarch: python.

Comment (ocefpaf) [reply recipes/phasefieldx/meta.yaml]: > I have added the suggested changes again, but I am getting this error (if I remove noarch, I also get errors)

For clarity the errors in the logs are:

```
Encountered problems while solving:
  - nothing provides requested fenics-dolfinx 0.8.0.*
  - nothing provides requested mpich
  - nothing provides requested petsc4py
  - nothing provides requested time
```

It is OK for noarch package to fails on Windows with a missing dependency. We prefer noarch b/c it is faster to build and the package will be ready when the missing dependency is available on Windows.

File: recipes/phasefieldx/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+source:
+  url: https://github.com/CastillonMiguel/phasefieldx/archive/v{{ version }}.tar.gz
+  sha256: f434c699c92575ed79576f7b266631e4e8bfc4c2b2b0d940b9efb715361339c6
+
+build:
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
+  number: 0
+  skip: True  # [win]
```
Comment (ocefpaf) [inline recipes/phasefieldx/meta.yaml:16]:
```suggestion
```

File: recipes/phasefieldx/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+  home: https://github.com/CastillonMiguel/phasefieldx
+  summary: 'PhaseFieldX: An Open-Source Framework for Advanced Phase-Field Simulations'
+  license: MIT
+  license_family: MIT
+  license_file: LICENSE
+  doc_url: https://phasefieldx.readthedocs.io
+  dev_url: https://github.com/CastillonMiguel/phasefieldx
+
+extra:
+  feedstock-name: phasefieldx
```
Comment (ocefpaf) [inline recipes/phasefieldx/meta.yaml:53]: Only needed if this is different from  the package name or the package has multiple outputs.

```suggestion
```

Comment (CastillonMiguel) [reply recipes/phasefieldx/meta.yaml]: Okay, I understand. I accept the commit to remove the line. Thank you!

**Pull request merged**